### PR TITLE
refactor: refactor unexported pack methods

### PR DESCRIPTION
**Pull request by Wwwsylvia**
I'm splitting PR #572 into 2 parts, this is the first part.

### CODE REVIEW

File: pack.go [hunk 1]
```diff
@@ -100,12 +100,12 @@ type PackManifestOptions struct {
...
+//   - If packManifestVersion is [PackManifestVersion1_1_RC4]:
 //     artifactType MUST NOT be empty unless opts.ConfigDescriptor is specified.
-//   - If packManifestVersion is [PackManifestVersion1_0],
-//     artifactType will be used as the the config media type unless
-//     opts.ConfigDescriptor is specified. If artifactType is empty,
-//     "application/vnd.unknown.config.v1+json" will be used.
+//   - If packManifestVersion is [PackManifestVersion1_0]:
+//     when opts.ConfigDescriptor is nil, artifactType will be used as the
+//     config media type; when opts.ConfigDescriptor is NOT nil,
+//     artifactType will be ignored.
```
Comment (shizhMSFT) [inline pack.go:108]: The new documentation is not accurate. If `opts.ConfigDescriptor` is `nil` and `artifactType` is empty, "application/vnd.unknown.config.v1+json" is used.

File: pack.go [hunk 2]
```diff
@@ -233,6 +228,50 @@ func packManifestV1_1_RC2(ctx context.Context, pusher content.Pusher, configMedi
...
 		Layers:      opts.Layers,
+		Annotations: annotations,
+	}
+	return pushManifest(ctx, pusher, manifest, manifest.MediaType, manifest.Config.MediaType, manifest.Annotations)
+}
+
+// packManifestV1_1_RC2 packs an image manifest as defined in image-spec
+// v1.1.0-rc2.
+// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md
+func packManifestV1_1_RC2(ctx context.Context, pusher content.Pusher, configMediaType string, layers []ocispec.Descriptor, opts PackOptions) (ocispec.Descriptor, error) {
```
Comment (shizhMSFT) [inline pack.go:239]: Let's see if we can do some refactoring in the next PR.

File: pack.go (hunk 1 above)
Comment (Wwwsylvia) [reply pack.go:108]: I was wondering if it's worth mention the empty artifactType behavior. We don't encourage users to leave `artifactType` empty, right?

Comment (shizhMSFT) [reply pack.go:108]: Yes, we don’t encourage users to have empty `artifactType`. However, we need to be honest to documenting the behavior if the users insist to leave it empty.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**